tests/foreman/api/test_classparameters.py

```diff
@@ -184,13 +324,26 @@ def test_negative_update_parameter_type(self):
 
         1. Parameter not updated with string type for invalid value.
         2. Error raised for invalid default value.
-
-        @caseautomation: notautomated
         """
+        sc_param = self.sc_params_list.pop()
+        for test_data in invalid_sc_parameters_data():
+            with self.subTest(test_data):
+                with self.assertRaises(HTTPError) as context:
```

Comment:
@renzon Good advice, thanks :)
However, in this particular case nested context managers are required because of assertRaises. There are 3 asserts, including assertRaises itself, that should be on the same level under subTest. Otherwise assertRaises will stop execution on step that raises exception and return control to the for loop, other asserts will be ignored.